kidviz/views.py

```diff
@@ -553,7 +553,15 @@ def get_context_data(self, **kwargs):
 
                             star_matrix_by_class[construct][observation.course][student][sublevel].append(observation)
 
-        min_date = Observation.get_min_date_from_observation(star_chart_4_obs) - datetime.timedelta(days=1)
+        min_date = Observation.get_min_date_from_observation(star_chart_4_obs)
+        max_date = Observation.get_max_date_from_observations(star_chart_4_obs)
+
+        if min_date:
```

Comment:
You can move this logic into `get_min_date_from_observation`. Also, probably you should use `min_time` with `combine` from `datetime` module